test/features/router/misc.js

```diff
@@ -84,4 +86,92 @@ describe('router - misc', function() {
         });
     });
 
+    it('should correctly resolve request templates', function() {
+        var requestTemplate = {
+            uri: '/{domain}/test',
+            method: 'post',
+            headers: {
+                'name-with-dashes': '{name-with-dashes}',
+                'global-header': '{$.request.params.domain}',
+                'added-string-header': 'added-string-header'
+            },
+            query: {
+                'simple': '{simple}',
+                'added': 'addedValue',
+                'global': '{$.request.headers.name-with-dashes}'
+            },
+            body: {
+                'object': '{object}',
+                'global': '{$.request.params.domain}',
+                'added': 'addedValue',
+                'nested': {
+                    'a': {
+                        'b': {
+                            c: '{a.b.c}'
```

Comment:
Could perhaps use a different depth and names, e.g. `{one.two}`, so that we are sure the mapping is not done by depth equality.